### PR TITLE
Use 'go generate' in bytesconv.go

### PR DESCRIPTION
**Pull request by ZYunH**
There are lots of tables in bytesconv.go, maybe we could use 'go generate' to generate these tables in a new file, called 'bytesconv_table.go'. In this way we can reduce init-time and also speed up our conversion, since the table is a constant which defined in compile-time.(feel free to modify my words if there is any syntax error)

### CODE REVIEW

File: bytesconv_table_gen.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	"io/ioutil"
+	"log"
+)
+
+const (
+	toLower = 'a' - 'A'
+)
+
+func main() {
+	var hex2intTable = func() [256]byte {
```
Comment (erikdubbelboer) [inline bytesconv_table_gen.go]: Can you change these to `:=` declarations like `hex2intTable := ...`. I prefer those, thanks!

File: bytesconv_table_gen.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	w := new(bytes.Buffer)
+	w.WriteString(pre)
+	for _, k := range tables {
+		v := m[k]
+		tmp := make([]byte, 256)
+		for i := 0; i < 256; i++ {
+			tmp[i] = v[i]
+		}
+		fmt.Fprintf(w, table, k, string(tmp))
+	}
```
Comment (erikdubbelboer) [inline bytesconv_table_gen.go]: I would just replace the whole above code with:
```go
fmt.Fprintf(w, "const hex2intTable = %q\n", hex2intTable)
fmt.Fprintf(w, "const toLowerTable = %q\n", toLowerTable)
fmt.Fprintf(w, "const toUpperTable = %q\n", toUpperTable)
fmt.Fprintf(w, "const quotedArgShouldEscapeTable = %q\n", quotedArgShouldEscapeTable)
fmt.Fprintf(w, "const quotedPathShouldEscapeTable = %q\n", quotedPathShouldEscapeTable)
```

File: bytesconv_table_gen.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+	out, err := format.Source(w.Bytes())
+	if err != nil {
+		log.Fatal(err)
+	}
```
Comment (erikdubbelboer) [inline bytesconv_table_gen.go]: This step isn't really needed. The generated source is so simple and easy to get right that formatting it shouldn't change it anymore.

File: bytesconv_table_gen.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+	if err := ioutil.WriteFile("bytesconv_table.go", out, 0660); err != nil {
+		log.Fatal(err)
+	}
+
```
Comment (erikdubbelboer) [inline bytesconv_table_gen.go]: Remove empty line.

File: bytesconv_table_gen.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+const pre = `// Code generated by go run bytesconv_table_gen.go; DO NOT EDIT.
+// See bytesconv_table_gen.go for more information about these tables.
+
+package fasthttp
+
+`
```
Comment (erikdubbelboer) [inline bytesconv_table_gen.go]: Can you move this constant to the top of the file, I prefer to have them there. Thanks.

File: bytesconv_table_gen.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+const table = `const %s = %#v
+`
```
Comment (erikdubbelboer) [inline bytesconv_table_gen.go]: This one can then be removed.

**Comment by ZYunH** [comment]
Thanks for review! all done

**Comment by erikdubbelboer** [comment]
Thanks!

**Pull request merged**